### PR TITLE
Woptim/shared ci v2023 12 0

### PR DESCRIPTION
**Pull request by adrienbernede**
### Summary

This PR updates RADIUSS Shared CI to improve flux allocations (controlled overlapping, support MPI tests).

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -71,7 +75,7 @@ stages:
     include:
       - local: '.gitlab/custom-jobs-and-variables.yml'
       - project: 'radiuss/radiuss-shared-ci'
-        ref: 'v2023.10.0'
+        ref: 'woptim/reproducer'
```
Comment (MrBurmark) [inline .gitlab-ci.yml:78]: Is this going to be updated in for the next release?

Comment (adrienbernede) [reply .gitlab-ci.yml:78]: @MrBurmark looks like I forgot to push my latest commit. Should be pointing at v2023.12.0

**Comment by MrBurmark** [comment]
Is this required for https://github.com/LLNL/RAJAPerf/pull/332?

**Comment by adrienbernede** [comment]
@MrBurmark, it will make using MPI in testing more robust, yes.

**Review by MrBurmark — APPROVED** [review]

**Comment by MrBurmark** [comment]
Looks like poodle timed out here as well, do we just need to add more time, or are there other factors?

**Comment by rhornung67** [comment]
> Looks like poodle timed out here as well, do we just need to add more time, or are there other factors?

Machine was down earlier. It came back, but runners were offline. I pinged LC. Runners are back and CI is running.

**Comment by adrienbernede** [comment]
@MrBurmark @rhornung67 I may have set the allocations a little to tight on ruby and poodle, given the recurring time outs.
Looking for a better solution to the root cause, I instead realized that I had no reason to specify the duration of the inner allocations, the outer one should suffice. After testing, it’s working well.

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**